src/server/debug/server/server.go

```diff
@@ -105,39 +109,60 @@ func (s *debugServer) handleRedirect(
 							return err
 						}
 						return collectDebugStream(tw, r)
-
 					}
 					pod, err := s.env.GetKubeClient().CoreV1().Pods(s.env.Config().Namespace).Get(ctx, f.Worker.Pod, metav1.GetOptions{})
 					if err != nil {
 						return errors.EnsureStack(err)
 					}
 					return s.handleWorkerRedirect(ctx, tw, pod, collectWorker, redirect)
+				case *debug.Filter_Database:
+					if err := s.collectDatabaseStats(ctx, tw, postgresPrefix); err != nil {
+						return err
+					}
 				}
 			}
-			// No filter, collect everything.
-			if err := collectPachd(ctx, tw, pachClient, pachdContainerPrefix); err != nil {
-				return err
-			}
-			pipelineInfos, err := pachClient.ListPipeline(true)
-			if err != nil {
-				return err
-			}
-			for _, pipelineInfo := range pipelineInfos {
-				if err := s.handlePipelineRedirect(ctx, tw, pachClient, pipelineInfo, collectPipeline, collectWorker, redirect); err != nil {
-					return err
-				}
-			}
-			if wantAppLogs {
-				// All other pachyderm apps (console, pg-bouncer, etcd, etc.).
-				if err := s.appLogs(ctx, tw); err != nil {
-					return err
-				}
-			}
-			return nil
+
+			return s.collectEverything(pachClient, collectPachd, collectPipeline, collectWorker, redirect, wantAppLogs,
+				ctx, tw, pachdContainerPrefix)
 		})
 	})
 }
 
+func (s *debugServer) collectEverything(
+	pachClient *client.APIClient,
+	collectPachd collectFunc,
+	collectPipeline collectPipelineFunc,
+	collectWorker collectWorkerFunc,
+	redirect redirectFunc,
+	wantAppLogs bool,
+	ctx context.Context,
+	tw *tar.Writer,
+	pachdContainerPrefix string,
+) error {
+	if err := collectPachd(ctx, tw, pachClient, pachdContainerPrefix); err != nil {
+		return err
+	}
+	pipelineInfos, err := pachClient.ListPipeline(true)
+	if err != nil {
+		return err
+	}
+	for _, pipelineInfo := range pipelineInfos {
+		if err := s.handlePipelineRedirect(ctx, tw, pachClient, pipelineInfo, collectPipeline, collectWorker, redirect); err != nil {
+			return err
+		}
+	}
+	if wantAppLogs {
+		// All other pachyderm apps (console, pg-bouncer, etcd, etc.).
+		if err := s.appLogs(ctx, tw); err != nil {
+			return err
+		}
+	}
+	if err := s.collectDatabaseStats(ctx, tw, postgresPrefix); err != nil {
```

Comment:
This probably needs a bool to control, so that it only runs on the pachd dump and not the workers.  This code is exceedingly confusing, but "collectPachd" and "collectPipeline" is basically a no-op sometimes.

Reply:
Ahh thanks for the explanation, that makes sense